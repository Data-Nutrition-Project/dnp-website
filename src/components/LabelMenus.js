import React from "react"

const LabelMenus = props => {
  return (
    <div>
      <header></header>
      <span>
        <p></p>
      </span>
      <p></p>
    </div>
  )
}

export default LabelMenus
