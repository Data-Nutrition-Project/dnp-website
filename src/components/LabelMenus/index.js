import React, { Component } from "react"
import { connect } from "react-redux"
import { sendBaseInfo } from "../../store/bases"
import PropTypes from "prop-types"
import styles from "./styles.module.css"
import classNames from "classnames"

const menus = [
  {
    title: "Overview",
    desc:
      "Overview information about the dataset including Description, Provenance, Composition, and Collection.",
  },
  {
    title: "Use Cases and Alerts",
    desc:
      "Relevant alerts for data practitioners who intent to use this dataset for specific use cases (types of analyses).",
  },
  {
    title: "Maintenance",
    desc:
      "Information about the ongoing management of the dataset, such as how the data will be maintained, updated, and the best contact for further inquiries.",
  },
]

class LabelMenus extends Component {
  constructor(props) {
    super(props)
    this.state = {
      showHighlight: this.props.highlightValue,
    }
    this.selectItem = this.selectItem.bind(this)
  }

  selectItem = item => {
    this.setState({
      selectedItem: item,
      showHighlight: false,
    })
    this.props.sendBaseInfo(item)
  }

  render() {
    return (
      <div
        className={styles.labelMenu}
        style={{ display: this.state.titleList }}
      >
        {menus.map((menu, id) => {
          return (
            <div
              id="selectButton"
              onClick={() => this.selectItem(menu.title)}
              className={classNames(styles.labelMenus, {
                [styles.highlightLabel]: this.state.selectedItem === menu.title,
              })}
              key={id}
            >
              <div
                className={classNames(styles.hiddenDiv, {
                  [styles.blockDiv]: this.state.showHighlight === true,
                })}
              ></div>
              <span className={styles.menuTitle}>{menu.title}</span>
              <div className={styles.flexbox}>
                <span className={styles.weightedLine}></span>
              </div>
              <p className={styles.menuParagraph}>{menu.desc}</p>
            </div>
          )
        })}
      </div>
    )
  }
}

LabelMenus.propTypes = {
  sendBaseInfo: PropTypes.func.isRequired,
}

const mapDispatch = dispatch => {
  return {
    sendBaseInfo: status => dispatch(sendBaseInfo(status)),
  }
}
export default connect(null, mapDispatch)(LabelMenus)
