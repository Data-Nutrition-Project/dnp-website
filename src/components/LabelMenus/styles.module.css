.labelMenu {
    width: 20%;
}

.labelMenus {
    height: 13rem;
    border: solid 0.0625rem #707070;
    background-color: #f4f4f4;
    margin: 2.1875rem 0rem;
    padding: 1.875rem 0rem 0rem 0rem;
}
.highlightLabel {
    background-image: url("/highlightbar.png");
    background-size: 0.5rem 35rem;
    background-repeat: no-repeat;
}

.labelMenuDescription {
    flex: 1;
    display: flex;
    width: 5.125rem;
    height: 0.875rem;
    font-family: Helvetica;
    font-size: 0.875rem;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: 1;
    letter-spacing: normal;
    text-align: left;
    margin: 0.25rem 0rem 0rem 1.46875rem;
}

.weightedLine {
    flex: 80% 0 0;
    border-bottom: 0.1875rem solid currentColor;
    display: inline;
    line-height: 0.85;
    margin: 0.46875rem 0rem 0rem 1.46875rem;
}

.menuParagraph {
    width: 15.375rem;
    height: 5.25rem;
    font-family: Helvetica;
    font-size: 0.75rem;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: 1;
    letter-spacing: normal;
    text-align: left;
    margin-left: 1.46875rem;
    margin-top: 0.53125rem;
}

.menuTitle {
    margin-left: 1.34375rem;
    width: 6.125rem;
    height: 1.375rem;
    font-family: Helvetica;
    font-size: 1.375rem;
    font-weight: bold;
    font-stretch: normal;
    font-style: normal;
    line-height: 1;
    letter-spacing: normal;
    text-align: left;
}

.flexbox {
    display: -webkit-flex;
    align-content: center;
    flex-wrap: wrap;
}

.flexComponents {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}
