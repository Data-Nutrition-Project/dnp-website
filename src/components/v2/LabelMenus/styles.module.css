@value layout: "../../layout.css";

@value tabletMaxWidth from layout;
@value landscapeMaxWidth from layout;
@value portraitMaxWidth from layout;

.menuTop {
  margin-bottom: 2rem;
}

.labelMenu {
  padding: 1.875rem 0 0 0;
  text-align: right;
  cursor: pointer;
  display: inline-block;
  position: relative;
  width: 100%;
}

.highlightMenuTitle {
  background-color: black;
  font-weight: bold;
  color: white;
  cursor: pointer;
}

.span {
  display: block;
  width: 100%;
}

.menuTitle {
  display: inline-block;
  margin-top: 0.5rem;
  font-size: 1.375rem;
  font-stretch: normal;
  font-style: normal;
  line-height: 1;
  letter-spacing: normal;
  text-align: right;
  padding: 0.5rem 2rem;
}

.menuTitle:hover {
  cursor: pointer;
}
.useButton .btn-primary {
  background-color: none !important;
}

@media (max-width: tabletMaxWidth) {
  .menuTitle {
    font-size: 0.8rem;
  }
}

@media (max-width: landscapeMaxWidth) {
  .menuTitle {
    font-size: 0.8rem;
  }

  .menuTop {
    margin-bottom: 1rem;
  }

  .labelMenu {
    display: flex;
    justify-content: space-around;
    padding: 0;
  }
}
