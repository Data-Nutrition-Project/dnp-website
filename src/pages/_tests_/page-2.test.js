import React from "react"
import { shallow } from "enzyme"
import renderer from "react-test-renderer"
import ShallowRenderer from "react-test-renderer/shallow"
import LabelMenus from "../page-2"
import { TestScheduler } from "jest"

describe("Test second page render", () => {
  it("renders correctly", () => {
    const renderer = new ShallowRenderer()
    const tree = renderer.render(<LabelMenus />)

    expect(tree).toMatchSnapshot()
  })
})
