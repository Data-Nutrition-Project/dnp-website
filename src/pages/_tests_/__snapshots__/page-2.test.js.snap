// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test second page render renders correctly 1`] = `<LabelWrapper />`;
